boolean step type failure (no attribute 'required')

Env:
```bash
$ which conjure-up
/snap/bin/conjure-up

$ conjure-up --version
conjure-up 2.5-alpha1

$ snap info conjure-up
name:      conjure-up
summary:   Package runtime for conjure-up spells
publisher: canonical
contact:   https://github.com/conjure-up/conjure-up
description: |
  This package provides conjure-up, an interface to installing spells that
  provide the user with an end to end walkthrough experience for installing and
  using big software.
snap-id: ijQR4HjYlTAmIjELJm5XCpNAqbDZWJ3h
commands:
  - conjure-up.conjure-down
  - conjure-up
  - conjure-up.juju
  - conjure-up.juju-wait
tracking:    edge
installed:   2.5-alpha1-20171130.1811 (856) 92MB classic
refreshed:   2017-11-30 12:26:11 -0600 CST
channels:                                   
  stable:    2.4.1-20171121.0211      (837) 82MB classic
  candidate: 2.4.2-20171130.1811      (859) 82MB classic
  beta:      2.4.2-20171130.1811      (859) 82MB classic
  edge:      2.5-alpha1-20171130.1811 (856) 92MB classic
```

I'm trying to use a `boolean` step type to present an option to my users.  The metadata looks like this, `./steps/05_logging/metadata.yaml` (note **required: True**):

```yaml
title: Logging Mechanism
description: Optional logging components
required: True
viewable: True
additional-input:
  - label: Elasticsearch/Filebeat/Graylog
    key: LOGGINGPLUGIN
    type: boolean
    default: "No"
```

You can see it looks nice:

![screen shot 2017-11-30 at 2 38 50 pm](https://user-images.githubusercontent.com/4576822/33454164-9dfef5b2-d5dd-11e7-9c90-1b6da897f45b.png)

As soon as I click 'Next' on this step (which recall, is **step 05_logging**), I get this error:

![screen shot 2017-11-30 at 2 39 14 pm](https://user-images.githubusercontent.com/4576822/33454195-baff1584-d5dd-11e7-9e44-00c243f6bed8.png)

Dunno if it matters, but note the bottom of the screen says it's running the `after-input` from **step 01_select-network**.  Error log snippet:

```
2017-11-30 14:38:19,850 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Padding selectable flow widget <AttrMap selectable flow widget <PlainButton selectable flow widget 'Next'> attr_map={None: 'button_primary'} focus_map={None: 'button_primary focus'}> align='right' width=('relative', 20)>, ('weight', 1))
2017-11-30 14:38:19,851 [DEBUG] conjure-up/canonical-kubernetes - widgets.input: input.py:159 - <RadioButton selectable flow widget 'flannel' state=False>
2017-11-30 14:38:19,851 [DEBUG] conjure-up/canonical-kubernetes - widgets.input: input.py:159 - <RadioButton selectable flow widget 'calico' state=False>
2017-11-30 14:38:19,851 [DEBUG] conjure-up/canonical-kubernetes - widgets.input: input.py:159 - <RadioButton selectable flow widget 'flannel' state=True>
2017-11-30 14:38:19,851 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Text flow widget ''>, ('weight', 1))
2017-11-30 14:38:19,852 [DEBUG] conjure-up/canonical-kubernetes - widgets.input: input.py:159 - <RadioButton selectable flow widget 'flannel' state=False>
2017-11-30 14:38:19,852 [DEBUG] conjure-up/canonical-kubernetes - widgets.input: input.py:159 - <RadioButton selectable flow widget 'calico' state=False>
2017-11-30 14:38:19,852 [DEBUG] conjure-up/canonical-kubernetes - widgets.input: input.py:159 - <RadioButton selectable flow widget 'flannel' state=True>
2017-11-30 14:38:19,852 [INFO] conjure-up/canonical-kubernetes - step.py:204 - Running step: 01_select-network after-input.
2017-11-30 14:38:20,062 [DEBUG] conjure-up/canonical-kubernetes - step.py:241 - Storing environment
2017-11-30 14:38:20,074 [DEBUG] conjure-up/canonical-kubernetes - step.py:247 - Executing script: /tmp/test/canonical-kubernetes/steps/01_select-network/after-input
2017-11-30 14:38:21,011 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Padding selectable flow widget <AttrMap selectable flow widget <PlainButton selectable flow widget 'Next'> attr_map={None: 'button_primary'} focus_map={None: 'button_primary focus'}> align='right' width=('relative', 20)>, ('weight', 1))
2017-11-30 14:38:21,026 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Text flow widget ''>, ('weight', 1))
2017-11-30 14:39:07,532 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Padding selectable flow widget <AttrMap selectable flow widget <PlainButton selectable flow widget 'Next'> attr_map={None: 'button_primary'} focus_map={None: 'button_primary focus'}> align='right' width=('relative', 20)>, ('weight', 1))
2017-11-30 14:39:07,532 [DEBUG] conjure-up/canonical-kubernetes - widgets.input: input.py:159 - <RadioButton selectable flow widget 'Yes' state=False>
2017-11-30 14:39:07,533 [DEBUG] conjure-up/canonical-kubernetes - widgets.input: input.py:159 - <RadioButton selectable flow widget 'No' state=False>
2017-11-30 14:39:07,533 [DEBUG] conjure-up/canonical-kubernetes - widgets.input: input.py:159 - <RadioButton selectable flow widget 'Yes' state=False>
2017-11-30 14:39:07,533 [DEBUG] conjure-up/canonical-kubernetes - widgets.input: input.py:159 - <RadioButton selectable flow widget 'No' state=True>
2017-11-30 14:39:07,534 [DEBUG] conjure-up/canonical-kubernetes - events.py:53 - Setting Error at conjureup/events.py:148
2017-11-30 14:39:07,538 [DEBUG] conjure-up/canonical-kubernetes - requests.packages.urllib3.connectionpool: connectionpool.py:207 - Starting new HTTP connection (1): www.google-analytics.com
2017-11-30 14:39:07,535 [ERROR] conjure-up/canonical-kubernetes - events.py:160 - Unhandled exception in <Task finished coro=<StepForm.validate() done, defined at /snap/conjure-up/856/lib/python3.6/site-packages/conjureup/ui/widgets/step.py:193> exception=AttributeError("'StepModel' object has no attribute 'required'",)>
Traceback (most recent call last):
  File "/snap/conjure-up/856/lib/python3.6/site-packages/conjureup/ui/widgets/step.py", line 206, in validate
    if not field.input.value and self.model.required:
AttributeError: 'StepModel' object has no attribute 'required'
2017-11-30 14:39:07,550 [DEBUG] conjure-up/canonical-kubernetes - __init__.py:21 - Showing dialog for exception: 'StepModel' object has no attribute 'required'
```

Full log at https://pastebin.canonical.com/204535/

--- a/conjureup/models/step.py
+++ b/conjureup/models/step.py
@@ -57,6 +57,7 @@ class StepModel:
         self.description = step.get('description', '')
         self.result = ''
         self.viewable = step.get('viewable', False)
+        self.required = step.get('required', False)
         self.needs_sudo = step.get('sudo', False)
         self.additional_input = step.get('additional-input', [])
         self.cloud_whitelist = step.get('cloud-whitelist', [])

